### PR TITLE
f-content-cards@0.11.0: Add skeleton loading card

### PR DESCRIPTION
**Pull request by asherstoppard**
First of two PR's to introduce skeleton loading state to the content card component

## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]

## Recording

ezgif-2-1758f54b0cb6

### CODE REVIEW

**Review by minrwhite — DISMISSED** [review]

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 1]
```diff
@@ -585,4 +585,83 @@ export default {
...
+            flex-flow: wrap;
+            flex-direction: row;
+        }
+    }
+    .offers-preloading {
+        border-radius: 4px;
+        display: flex;
+        flex-direction: column;
+        flex: 0 0 40%;
+        margin: 0 8px 24px 0;
```
Comment (simonsandersje) [inline packages/f-content-cards/src/components/ContentCards.vue]: spacing() for padding, but not for margin?

**Review by simonsandersje — DISMISSED** [review]
LGTM, with one nit-picking comment!

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 2]
```diff
@@ -585,4 +585,83 @@ export default {
...
+        margin-top: 40px;
+
+        @include media('>=narrowMid') {
+            display: flex;
+            flex-flow: wrap;
+            flex-direction: row;
+        }
+    }
+    .offers-preloading {
+        border-radius: 4px;
```
Comment (xander-marjoram) [inline packages/f-content-cards/src/components/ContentCards.vue]: Nit: Is there a fozzie variable you can use for this?

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 1 above)
Comment (asherstoppard) [reply packages/f-content-cards/src/components/ContentCards.vue]: @simonsandersje Changed those over to use `spacing()` :+1:

File: packages/f-content-cards/src/components/ContentCards.vue (hunk 2 above)
Comment (asherstoppard) [reply packages/f-content-cards/src/components/ContentCards.vue]: @xander-marjoram There is, there's also a variable set as part of the component so I've switched it over to that :+1:

**Review by minrwhite — DISMISSED** [review]

File: packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<template>
+    <div>
+        <div class="offers-preloading-wrap">
```
Comment (tommcclean) [inline packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: Couple of NIT comments:
- The extra div wrapper could probably be removed(?)
- Should the CSS for offers-preloading-wrap not be in this SFC?

File: packages/f-content-cards/CHANGELOG.md [hunk 4]
```diff
@@ -3,13 +3,23 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+
+v0.11.0
+------------------------------
+*July 9, 2020*
```
Comment (ashleynolan) [inline packages/f-content-cards/CHANGELOG.md]: Probably worth bumping this as it's quite out of date.

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 5]
```diff
@@ -585,4 +585,83 @@ export default {
...
+
+        @include media('>=narrowMid') {
+            flex-direction: row;
+            max-width: 370px;
+            flex: 0 0 40%;
+        }
+    }
+
+    .offers-preloading:empty {
+        height: 403px;
```
Comment (ashleynolan) [inline packages/f-content-cards/src/components/ContentCards.vue]: Bit of a magic number – is there a reason it's exactly `403px`? If so, probably worth a comment.

File: packages/f-content-cards/src/components/ContentCards.vue [hunk 6]
```diff
@@ -585,4 +585,83 @@ export default {
...
+        background-color: $white;
+
+        background-image: radial-gradient(0 at 0 0, lightgray 99%, transparent 0),
+        linear-gradient(100deg, rgba(255, 255, 255, 0) 0%,
```
Comment (ashleynolan) [inline packages/f-content-cards/src/components/ContentCards.vue]: Can you tab these inline with the radial-gradient statement above it, or does the linter complain about doing it?

Also, would be worth a comment as to what sort of image this produces (as it's hard to imagine from just looking at gardients).

File: packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue (hunk 3 above)
Comment (asherstoppard) [reply packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: Yeah you're absolutely right :+1: I'll get those changes pushed.

Comment (asherstoppard) [reply packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: @tommcclean Pushed a change to address applying the styles in the SFC but utilising modules breaks the storybook integration. There's a separate change to modularise all styling across the component so I'll investigate that issue and encapsulate the styling as part of that ticket (CACT-583) if that's ok with you?

**Review by kevinrodrigues — DISMISSED** [review]

**Comment by DamianMullins** [comment]
@asherstoppard Are you able to add some screenshots/video to the PR description, please?

File: packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        count: {
+            type: Number,
+            default: 1
+        }
+    }
+};
+</script>
+
+<style lang="scss">
+    .offers-preloading-wrap {
```
Comment (DamianMullins) [inline packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: Not sure what the deal is with class names in components when they aren't declared as modules, but typically component classes would start with "c-" and any subsequent dashes signifying child elements.

@ashleynolan Is this something you've considered before? I'm guessing we should adhere to the naming conventions when the CSS can affect other styles on a page, but when we use module we can be more relaxed?

**Review by minrwhite — DISMISSED** [review]

**Review by minrwhite — DISMISSED** [review]

File: packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue [hunk 8]
```diff
@@ -0,0 +1,108 @@
+<template>
+    <div class="c-contentCard-skeletonLoader-wrapper">
```
Comment (ashleynolan) [inline packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: If this is the base of the `c-contentCard-skeletonLoader` classes, you can lose the `-wrapper` part of the classname here.

Comment (ashleynolan) [reply packages/f-content-cards/src/components/cardTemplates/SkeletonLoader.vue]: Also, when this is used – is it inside a `c-contentCard` element? If not, then it should be named more like `c-skeletonLoaderCard`

**Review by minrwhite — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**